script/main.py

```diff
@@ -19,23 +23,23 @@ def create_driver(name):
   return driver
 
 
-def list_of_all_links(driver):
-  links_elem = driver.find_elements_by_tag_name("a")
-  return links_elem  
+def find_links(driver):  
+  links_elem = driver.find_elements_by_tag_name("a")    
+  return [LinkInfo(name, name.get_attribute('href')) for name in links_elem]
```

Comment:
You can't use name variable here `links_elem` is not list of names. It is list of elements. And element contains href and also text.